plugins/woocommerce/client/legacy/js/frontend/checkout.js

```diff
@@ -8,6 +8,38 @@ jQuery( function( $ ) {
 
 	$.blockUI.defaults.overlayCSS.cursor = 'default';
 
+	const ajax = options => {
+		const controller = new AbortController();
+
+		window.fetch( options.url, {
+			method: options.type,
+			headers: { 'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8' },
+			body: options.data,
+			signal: controller.signal
+		} )
+			.then( response => {
+				response.text().then( text => {
+					if ( !response.ok ) {
+						const error = new Error( response.statusText );
+						error.responseText = text; // Needed for when wc_checkout_params.debug_mode is enabled
+						throw error;
+					}
+
+					if ( options.dataType === 'html' ) {
+						options.success( text );
+						return;
+					}
+
+					const json = JSON.parse( ajax.dataFilter( text, 'json' ) );
+					options.success( json );
+				} );
+			} )
+			.catch( error => options.error && options.error( error.responseText, 'error', error.message ) );
+
+		return controller;
+	};
+	ajax.dataFilter = ( raw_response, dataType ) => raw_response;
```

Comment:
This style of having a single mutable function attached to ajax reminds me of some of the old styles of doing JS (I see why you did it to make the functionality similar to jQuery). But I think we should adjust this code and instead use default args to pass in a `dataFilter`.

e.g.


```
const ajax = (options, dataFilter = (raw_response, dataType) => raw_response) => {
  ...
}
```

or it could become an extra property on `options`?

Reply:
I like the paramater to **ajax()** approach. I've gone with the mutable function to preserve the old (weird) behaviour where from the moment of calling **submit()**, every following **ajax()** call  - including the non-**submit()** ones! - would also use **dataFilter** (from my understanding). It is strange, yes, but I didn't want to further change any old behaviour.

Whad do you gather? Shall we change it to a param to **ajax()** regardless?

Reply:
Yikes, ok yes probably safer to leave this as is.